flink-architecture-tests/flink-architecture-tests-base/src/main/java/org/apache/flink/architecture/common/ImportOptions.java

```diff
@@ -57,4 +70,19 @@ public boolean includes(Location location) {
             return !location.matches(SHADED);
         }
     }
+
+    /** {@link Pattern} based predicate. */
+    private static class PatternPredicate implements Predicate<Location> {
```

Comment:
this class seems unnecessary. Why not just do it like `ExcludeShadedImportOption`.

Reply:
fair enough, thanks!